include/libtorrent/stack_allocator.hpp

```diff
@@ -79,8 +79,9 @@ namespace libtorrent { namespace aux
 		{
 			int const ret = int(m_storage.size());
 			int const len = int(std::strlen(str));
+			TORRENT_ASSERT(len > 0);
 			m_storage.resize(ret + len + 1);
-			std::strcpy(&m_storage[ret], str);
+			std::memcpy(&m_storage[ret], str, len);
```

Comment:
I still think it would be a good idea to make it explicit. Partly because it avoids questions like this and I can envision a future where ``std::vector<>`` is replaced by something that doesn't default initialize the memory (as an optimization)